mlflow/models/model.py

```diff
@@ -29,55 +30,124 @@
+        """
+        self._artifact_path = artifact_path
+        self._flavors = flavors
+        self._model_uri = model_uri
+        self._model_uuid = model_uuid
+        self._run_id = run_id
+        self._saved_input_example_info = saved_input_example_info
+        self.signature_dict = signature_dict
+        self._signature = signature
+        self._utc_time_created = utc_time_created
+        self._mlflow_version = mlflow_version
 
-    #: A dictionary mapping the flavor name to how to serve the model as that flavor. For example:
-    #:
-    #: .. code-block:: python
-    #:
-    #:     {
-    #:         "python_function": {
-    #:             "model_path": "model.pkl",
-    #:             "loader_module": "mlflow.sklearn",
-    #:             "python_version": "3.8.10",
-    #:             "env": "conda.yaml",
-    #:         },
-    #:         "sklearn": {
-    #:             "pickled_model": "model.pkl",
-    #:             "sklearn_version": "0.24.1",
-    #:             "serialization_format": "cloudpickle",
-    #:         },
-    #:     }
-    flavors: Dict[str, Any]
+    @property
+    def artifact_path(self):
+        return self._artifact_path
 
-    #: The ``model_uri`` of the logged model in the format ``'runs:/<run_id>/<artifact_path>'``.
-    model_uri: str
+    @property
+    def flavors(self):
+        return self._flavors
 
-    #: The ``model_uuid`` of the logged model, e.g., ``'39ca11813cfc46b09ab83972740b80ca'``.
-    model_uuid: str
+    @property
+    def model_uri(self):
+        return self._model_uri
 
-    #: The ``run_id`` associated with the logged model, e.g., ``'8ede7df408dd42ed9fc39019ef7df309'``
-    run_id: str
+    @property
+    def model_uuid(self):
+        return self._model_uuid
 
-    #: A dictionary that contains the metadata of the saved input example, e.g.,
-    #: ``{"artifact_path": "input_example.json", "type": "dataframe", "pandas_orient": "split"}``.
-    saved_input_example_info: Optional[Dict[str, Any]]
+    @property
+    def run_id(self):
+        return self._run_id
 
-    #: A dictionary that describes the model input and output generated by
-    #: :py:meth:`ModelSignature.to_dict() <mlflow.models.ModelSignature.to_dict>`.
-    signature_dict: Optional[Dict[str, Any]]
+    @property
+    def saved_input_example_info(self):
+        return self._saved_input_example_info
 
-    #: The UTC time that the logged model is created, e.g., ``'2022-01-12 05:17:31.634689'``.
-    utc_time_created: str
+    @property
+    def signature_dict(self):
+        return self._signature_dict
 
-    #: Version of MLFlow used to log the model
-    mlflow_version: str
+    @signature_dict.setter
+    def signature_dict(self, value):
+        # pylint: disable=attribute-defined-outside-init
+        warnings.warn(
+            "Field signature_dict is deprecated since v1.28.1. Use signature instead.",
+            category=DeprecationWarning,
```

Comment:
```suggestion
            category=FutureWarning,
```

Reply:
btw why do we raise a warning in a setter?

Reply:
Nope, if we put the warning into def signature_dict it doesn't actually warn when initiating a new instance. But if you just want to warn once we can also move this warning into __init__ though.

Reply:
We want to warn when users access the `signature_dict` property, right?

Reply:
Yes, I added another warning in def signature_dict so that both getter & setter warns.